File: src/AcWing/contest/4/B.py
n , k = map(int , input().split())

def dfs(n , k):
    if k == 1 << n -1:
        return n
    if k < 1 << n -1:
        return dfs(n - 1, k)
    return dfs(n - 1,k - (1 << n - 1))
print(dfs(n,k))

File: src/AcWing/contest/4/B.cpp
#include <bits/stdc++.h>

using namespace std;
typedef long long ll;
int n;
ll k;
ll dfs(int n,ll k){
    if(k == 1ll << n -1) return n;
    if(k < 1ll << n - 1) return dfs(n - 1,k);
    return dfs(n - 1, k - (1ll << n - 1));
}
int main()
{
    cin >> n >> k;
    cout << dfs(n,k) <<endl;
    return 0;
}